yt/frontends/boxlib/fields.py

```diff
@@ -523,15 +525,24 @@ def setup_fluid_fields(self):
 
 
 def _nice_species_name(field):
-    spec_match = spec_finder.search(field)
-    nice_name = "".join(spec_match.groups())
+    try:
+        spec_match = spec_finder.search(field)
+        nice_name = "".join(spec_match.groups())
+
+    except AttributeError as e:
+        # Catch exception cases for e.g. molecules (H2O, C6H6)
```

Comment:
Upon a closer look, I don't think we don't need a try/except at all here
```suggestion
    if (spec_match := spec_finder.search(field)) is not None:
        nice_name = "".join(spec_match.groups())
    else:
        # molecules (e.g. H2O, C6H6)
```